### DISCUSSION TITLE
How to export mesh for Victoria 3

### DISCUSSION THREAD

**Discussion by Thefarmnored** [ross-g/io_pdx_mesh#88] [Q&A] [answered]
Hello, about this plugin, I encountered a few points in the use, the first is the relevant environment when I encountered problems
System version: Windows 10
Maya version: 2018.3
Blende versionr: 3.6.5
Victoria 3 version: 1.4
io_pdx_mesh version: 0.9

Victoria3 mesh files seem to import normally in both Maya and Blender, but again, the exported mesh file does not seem to be recognized in Victoria3. When I replace the exported mesh file with the vanilla Victoria3 mesh file in debug mode, The model shows

I think Victoria 3 uses a new mesh file format, so how do I make the exported mesh file available in Victoria 3

**Comment by Thefarmnored** [comment]
I tried to use the CK3 model in Victoria 3, but again I couldn't see the model on the map, only when I replaced the model with the original mesh file of Victoria 3. So I thought, This may require first figure out what is the difference between V3 mesh and previous mesh

**Comment by Thefarmnored** [comment]
I may need to describe this in more detail

When I changed the mesh file in debug mode, I only changed the file with the file extension ".mesh", and the rest did not change. So this proves that the file structure I tested and the syntax in the code is in compliance with the specification, so the problem should be expanding the file named ".mesh", but I don't know how the exported mesh differs from the vanilla mesh of Victoria 3

**Comment by Thefarmnored** [comment]
[wonders.zip](https://github.com/ross-g/io_pdx_mesh/files/13325651/wonders.zip)
We would appreciate it if you would help us with this problem. Here are three vanilla Victoria 3 model files that I thought you might need it

**Comment by ross-g** [comment, marked as answer]
So I don't own Victoria 3 and hadn't seen any of its' assets before to experiment with. Happily there is a free play weekend on Steam at the moment, so I am able to look at their models and do some quick tests. The tool seems to works fine with Victoria 3 meshes/animation and from what I can see there weren't any format changes. 

A brief example, I made an edit so the Pyramids of Giza look silly.
![image](https://github.com/ross-g/io_pdx_mesh/assets/9258768/846cb2e8-8b94-4229-95d1-3b1efb29de1f)

I think you have missed some modding details, I won't go into too much depth as I don't want this to become a mod support forum but:
> You can't just copy, paste and rename a `.mesh` file from one Clausewitz game to another.
> You can't export your own model over an existing `.mesh` file.

Meshes aren't just the `.mesh` file, they are also the entity defined inside the  `.asset` file which tells the game what shaders and textures to use through defining mesh settings (**by name** of each sub-mesh contained in the `.mesh` file). So all of these mesh setting names will be wrong if you just copy and paste files or overwrite an existing file. To mod successfully you also need to learn how to edit the `.asset` file.

**Reply by Thefarmnored** [reply]
Sorry to bother you, I have found the reason, is that the texture file specified in the material property when exporting the `.mesh` file does not exist. So I'm pretty sure that there's no problem with this plugin at the moment.

**Reply by Thefarmnored** [reply]
More specifically, the path of the texture map is incorrect. If it's not the texture map path is specified in the `.mesh` file cannot be found, it's not the path of the problem map is not specified in the `.mesh` file.

When I was testing with the vanilla model, I temporarily removed the texture file from the vanilla model, and the model was not displayed, after I recover the texture map from the recycle bin, the model was displayed. So the answer is that even if the `.asset` file specifies the path to the texture map, the model's material properties still need to contain a valid path.

The fact is that the model can be displayed even if the `name` specified in the `.asset` file does not match the name of the mesh object in the `.mesh` file, provided that the `.mesh` file specifies the texture map that can be found, in which case the texture map displayed is the one that the model can find. So one conclusion is:

>>The model cannot be displayed if all three texture map cannot be found.

Regarding the names of mesh objects, I noticed that in vanilla, all the mesh objects in `.mesh` files have the same name, just like all the models in vanilla that needed to define coordinates in the path: `gfx/map/map_object_data`, these models have the map path specified in the `.mesh` file. My test result is that the name of the `.asset` file can be wrong, but the texture specified in the `.mesh` file must be found, otherwise it will not be displayed.

**Reply by Thefarmnored** [reply]
I also made a more obvious mistake earlier, when I tried to export a `.mesh` file, I used Blender in a version other than the recommended 3.3, and the Maya version I used was 2018.3, but I don't seem to have tried to export vanilla `.mesh` in Maya. It's my problem.

**Reply by Thefarmnored** [reply]
One detail is that in Blender's default Settings, the path to the texture map will only show the final image file name and its extension name, but I didn't know that earlier. When I saw that both the `.mesh` file and the `.dds` file were in the same path, I thought that was the actual path... But it's not.

**Reply by Thefarmnored** [reply]
Okay, maybe I should add one more detail. I did not receive your reply a few days ago, I just found out the cause of this problem today, so I wanted to send a new reply here to show that I was mistaken... But you have finished the test and gave a reply. I'm sorry that I found the reason a little late.

**Comment by Thefarmnored** [comment]
![Snipaste_2023-11-23_13-00-49](https://github.com/ross-g/io_pdx_mesh/assets/94956852/ed783b0a-d8ac-4b5f-b53b-a5e86878f2b0)
Since I didn't know how to see the full texture path in Blender, I decided to do it in Maya. This time, I successfully implemented the model display.

**Reply by Thefarmnored** [reply]
Of course, I obviously need to adjust the size of this model, I also need to complete the production of UV map, and I also need to adjust the rotation values in the locator.
